Move clipboard operation implementations to individual editor subscreens

bdach opened the pull request:
Closes #14756 by implementing [the proposal given therein](https://github.com/ppy/osu/issues/14756#issuecomment-957209214). This also paves the way for potentially having screen-specific clipboard functionality, like being able to copy & paste timing points. Was a pretty easy refactor and I think this came out pretty well.

The one thing I am not sure of is leaving editor-global `Clipboard` and resolving it in screens. I could move it down to the compose screen (which is the only user now), but that would mean that clipboard becomes screen-contextual. Which simplifies some things - no need to check whether data in clipboard is valid for a given screen (which is not a concern now but will become one later) - but also maybe complicates UX. Will apply the above on request.

One semi-accidental consequence of this change is that all clipboard operations are now disabled on all screens except compose. I think that makes sense, anyway - being able to copy an object on compose screen, move to timing and then paste an object in at that time via <kbd>Ctrl</kbd>+<kbd>V</kbd> is arguably a bug.

bdach committed:
Move clipboard operation implementation down to current screen

bdach committed:
Move clipboard action availability logic down to editor screens

bdach committed:
Fix test failures due to missing dependencies

bdach commented:
I actually remembered there was a second reason for leaving the `Clipboard` bindable in editor - namely the difficulty change flow and the way it preserves clipboard contents. That one is more difficult to get rid of, unfortunately...

peppy commented on osu.Game.Rulesets.Mania.Tests/Editor/TestSceneManiaComposeScreen.cs:
```diff
@@ -22,6 +23,9 @@ public class TestSceneManiaComposeScreen : EditorClockTestScene
         private SkinManager skins { get; set; }
 
+        [Cached(Name = nameof(Screens.Edit.Editor.Clipboard))]
+        private Bindable<string> clipboard = new Bindable<string>();
```
I think I'd feel 100% better if this was encapsulated in an object DI purposes.

```csharp
public class Clipboard
{
    public Bindable<string> Content {get;} = new Bindable<string>();
}
```

peppy commented on osu.Game/Screens/Edit/EditorScreen.cs:
```diff
@@ -49,5 +50,54 @@ protected override void PopOut()
+        /// <summary>
+        /// Performs a "cut to clipboard" operation appropriate for the given screen.
+        /// </summary>
+        protected virtual void PerformCut()
```
Wondering if it makes sense for the clipboard instance to be passed through here (and the other methods in this file) as a parameter? Should theoretically avoid any need to DI the clipboard.

bdach replied:
It's a little dicey here because the screens will `BindValueChanged` to the clipboard content to update the availability state of all the actions, so I'd still probably feel more comfortable with DI'ing.

bdach replied:
Unless I split `CanCut()` into methods and call that on clipboard content changes, but that feels less flexible than what is currently here (especially so that the action availability may depend on other context, like e.g. how many objects are selected - which the composer already uses).

peppy replied:
I'm not too fussed just yet. Let's go with what is here and see how it builds out.

bdach committed:
Extract class for clipboard contents for DI purposes

bdach committed:
Copy editor timestamp to OS clipboard even when triggered via menu bar

Would only work when triggered via Ctrl+C before, and not work at all
for Ctrl+X.

bdach commented:
Have also taken the liberty of unifying the existing "copy timestamp" flow in composer screen with the new screen-local clipboard flow. This makes it so copying via menu bar also will copy the timestamp out to OS, which it didn't use to do (only worked via Ctrl+C). Seemed like this pull is an appropriate place to do so anyhow.

peppy committed:
Merge branch 'master' into editor-clipboard-refactor
